explorer/src/components/common/VerifiedBadge.tsx

```diff
@@ -0,0 +1,91 @@
+import { PublicKey } from "@solana/web3.js";
+import { useEffect, useState } from "react";
+
+export type AnchorBuild = {
+  aborted: boolean;
+  address: string;
+  created_at: string;
+  updated_at: string;
+  descriptor: string[];
+  docker: string;
+  id: number;
+  name: string;
+  sha256: string;
+  upgrade_authority: string;
+  verified: string;
+  verified_slot: number;
+  state: string;
+};
+
+/**
+ * Returns a verified build from the anchor registry. null if no such
+ * verified build exists, e.g., if the program has been upgraded since the
+ * last verified build.
+ */
+async function getVerifiedBuild(
+  deploySlot: number,
+  programId: PublicKey,
+  limit: number = 5
+) {
+  const url = `https://anchor.projectserum.com/api/v0/program/${programId.toString()}/latest?limit=${limit}`;
+  const latestBuildsResp = await fetch(url);
+
+  // Filter out all non successful builds.
+  const latestBuilds = (await latestBuildsResp.json()).filter(
+    (b: AnchorBuild) =>
+      !b.aborted && b.state === "Built" && b.verified === "Verified"
+  ) as AnchorBuild[];
+
+  if (latestBuilds.length === 0) {
+    return null;
+  }
+
+  // Get the latest build.
+  const build = latestBuilds[0];
+
+  // Has the program been upgraded since the last build?
+  if (deploySlot !== build.verified_slot) {
+    return null;
```

Comment:
Can you add a special status in the UI for this case? Something like "Latest update unverified"

Reply:
This is a bit tricky since the distinction between "unverified" and "last update unverified" would be dependent on how many builds we poll from the history (by default 5 here). E.g. if there were 4 updates since the last verification it would show as "last update unverified", and after just 1 more unverified upgrade it would show as "unverified".
In summary the distinction might easily be confusing to the user. Wdyt?

Reply:
I think it's unlikely that there will be 5 consecutive builds without verification right? Or am I misunderstanding the anchor verification process?

Reply:
Added a separate row for the last verified deployment slot when available, as additional info for users.